### PR TITLE
Simplify wallet-core package

### PR DESCRIPTION
**Pull request by snario**
The `wallet-core` package had a lot of baggage inside of it from being ripped out of `xstate-wallet`. This PR prunes unnecessary linting configuration options, adds missing dependencies, and removes the emission of `.js` files from the package so it's simply a distribution of TypeScript files for now.

### CODE REVIEW

**Comment by snario** [comment]
TODO: Configure `webpack` of `xstate-wallet` to read TS files.

File: packages/wallet-core/src/bignumber.ts [hunk 1]
```diff
@@ -49,7 +49,8 @@ export class BN {
   static toHexString = unaryOperator('toHexString');
 
   static from = (n: BigNumberish | BN): Uint256 => EthersBigNumber.from(n).toHexString() as Uint256;
-  static isBigNumber = val => typeof val === 'string' && !!val.match(/^0x[0-9A-Fa-f]{0,64}$/);
+  static isBigNumber = (val: any) =>
```
Comment (andrewgordstewart) [inline packages/wallet-core/src/bignumber.ts:52]: Probably worth renaming this to `isUint256` -- maybe it's worth it to squeeze into this PR?

**Comment by snario** [comment]
Yeah. Will do

**Comment by snario** [comment]
Cancelled the TS distribution thing... was causing headaches.

**Pull request merged**